src/macrobrowserui.cpp

```diff
@@ -4,14 +4,16 @@
 #include <QJsonDocument>
 #include <QJsonArray>
 
+const int MacroBrowserUI::FileRole = Qt::UserRole + 0;
+const int MacroBrowserUI::UrlRole = Qt::UserRole + 1;
+const int MacroBrowserUI::PathRole = Qt::UserRole + 2;
+const int MacroBrowserUI::PopulatedRole = Qt::UserRole + 3;
+Q_DECLARE_METATYPE(QTreeWidgetItem *)
```

Comment:
why is the Declare metatype needed ?

Reply:
First I didn't use it, but the the linux build failed after posing the PR. Message said that the metatype macro should be used. I tried some workarounds but without success (sorry, that I already deleted all the CI/CD runs). Then I searched the manual and found an [enumeration](https://doc.qt.io/qt-6/qmetatype.html#Type-enum). The QtreeWidgetItem isn't in the list. So I used the macro.

Today I found similar code in bookmark.cpp (l. 115, 131) that might help:
```
QDocumentLineHandle *dlh = qvariant_cast<QDocumentLineHandle *>(item->data(DocLineHandle));
```
I just tried the following:
```
QTreeWidgetItem *currentItem =
             qvariant_cast<QTreeWidgetItem *>(reply->request().attribute(mbAttributeItem));
```
Same [error](https://github.com/texstudio-org/texstudio/actions/runs/8088743327/job/22103390150/#step:6:1):
```
In file included from /home/runner/work/texstudio/texstudio/src/modifiedQObject.h:5,
                 from /home/runner/work/texstudio/texstudio/src/mostQtHeaders.h:12,
                 from /home/runner/work/texstudio/texstudio/CMakeFiles/texstudio.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qmetatype.h: In instantiation of ‘constexpr int qMetaTypeId() [with T = QTreeWidgetItem*]’:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qvariant.h:371:37:   required from ‘static QVariant QVariant::fromValue(const T&) [with T = QTreeWidgetItem*]’
/home/runner/work/texstudio/texstudio/src/macrobrowserui.cpp:109:61:   required from here
/usr/include/x86_64-linux-gnu/qt5/QtCore/qmetatype.h:1916:5: error: static assertion failed: Type is not registered, please use the Q_DECLARE_METATYPE macro to make it known to Qt's meta-object system
 1916 |     Q_STATIC_ASSERT_X(QMetaTypeId2<T>::Defined, "Type is not registered, please use the Q_DECLARE_METATYPE macro to make it known to Qt's meta-object system");
      |     ^~~~~~~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtCore/qmetatype.h:1916:5: note: ‘false’ evaluates to false
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:54,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/QObject:1,
                 from /home/runner/work/texstudio/texstudio/src/modifiedQObject.h:35,
                 from /home/runner/work/texstudio/texstudio/src/mostQtHeaders.h:12,
                 from /home/runner/work/texstudio/texstudio/CMakeFiles/texstudio.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qmetatype.h: In instantiation of ‘static constexpr int QMetaTypeId2<T>::qt_metatype_id() [with T = QTreeWidgetItem*]’:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qmetatype.h:1917:43:   required from ‘constexpr int qMetaTypeId() [with T = QTreeWidgetItem*]’
/usr/include/x86_64-linux-gnu/qt5/QtCore/qvariant.h:371:37:   required from ‘static QVariant QVariant::fromValue(const T&) [with T = QTreeWidgetItem*]’
/home/runner/work/texstudio/texstudio/src/macrobrowserui.cpp:109:61:   required from here
/usr/include/x86_64-linux-gnu/qt5/QtCore/qmetatype.h:1776:96: error: ‘qt_metatype_id’ is not a member of ‘QMetaTypeId<QTreeWidgetItem*>’
 1776 |     static inline Q_DECL_CONSTEXPR int qt_metatype_id() { return QMetaTypeId<T>::qt_metatype_id(); }
      |                                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
make[2]: *** [CMakeFiles/texstudio.dir/build.make:9572: CMakeFiles/texstudio.dir/src/macrobrowserui.cpp.o] Error 1
```
So I would return to my solution when you have no other suggestion.